src/DiIiS-NA/D3-GameServer/GSSystem/ItemsSystem/AffixGenerator.cs

```diff
@@ -321,24 +319,28 @@ public static void Generate(Item item, int affixesCount, bool isCrafting = false
 		public static int FindSuitableAffix(Item item, int affixGroup, int affixLevel, bool extendedFilter)
 		{
 			if (affixGroup == -1) return -1;
-			var all_group = LegendaryAffixList.Where(a => (a.AffixFamily0 == affixGroup || a.AffixFamily1 == affixGroup) && (Item.Is2H(item.ItemType) ? !a.Name.EndsWith("1h") : (!a.Name.Contains("Two-Handed") && !a.Name.EndsWith("2h"))));
+			var allGroup = LegendaryAffixList
+				.Where(a => (a.AffixFamily0 == affixGroup || a.AffixFamily1 == affixGroup) && (Item.Is2H(item.ItemType) ? !a.Name.EndsWith("1h") : (!a.Name.Contains("Two-Handed") && !a.Name.EndsWith("2h"))))
+				.ToArray();
 
-			if (all_group.Count() == 0) return -1;
+			if (!allGroup.Any()) return -1;
 
-			bool secondGroup = (all_group.First().AffixFamily1 == affixGroup);
+			bool secondGroup = allGroup.First().AffixFamily1 == affixGroup;
 
-			var suitable = all_group.Where(a => a.AffixLevel <= affixLevel || affixLevel <= 0);
+			var suitable = allGroup.Where(a => a.AffixLevel <= affixLevel || affixLevel <= 0).ToArray();
 
-			if (suitable.Count() == 0) return -1;
+			if (!suitable.Any()) return -1;
 
 			List<int> itemTypes = ItemGroup.HierarchyToHashList(item.ItemType);
 
-			suitable = suitable.Where(a => itemTypes.ContainsAtLeastOne(a.ItemGroup) || (extendedFilter ? itemTypes.ContainsAtLeastOne(a.LegendaryAllowedTypes) : false));
+			suitable = suitable.Where(a =>
+				itemTypes.ContainsAtLeastOne(a.ItemGroup) ||
+				(extendedFilter && itemTypes.ContainsAtLeastOne(a.LegendaryAllowedTypes))).ToArray();
 
-			if (suitable.Count() == 0) return -1;
```

Comment:
By replacing `Count() == 0` with `!Any()` we do not iterate to the end

Reply:
if it's multiple iterations there then maybe make it an immutable array, as most of it it's List<T> or IEnumerable<T>, on the List<T> ones probably you won't find a place that it's required a List. I haven't, so far.